src/components/widgets/MatchHistory.tsx

```diff
@@ -0,0 +1,98 @@
+import { PlayersApi } from "../../lib/api"
+import { Suspense, createEffect, createResource, createSignal, on, onMount, useTransition } from "solid-js"
+import { Widget } from "../ui/Widget"
+import { Pagination } from "../ui/Pagination"
+import MatchPreview from "./MatchPreview.tsx"
+
+interface Props {
+  page?: number
+  limit?: number
+  playerId: string
+}
+const perPage = 20
+
+export function MatchHistory(props: Props) {
+  const [count, setCount] = createSignal(props.limit ?? perPage)
+
+  const [page, setPage] = createSignal(props.page || 1)
+
+  const [isPending, start] = useTransition()
+  const [isBrowserNavigation, setIsBrowserNavigation] = createSignal(true)
+
+  const playerId = props.playerId
+
+  const getOptions = () => ({
+    count: count(),
+    page: page(),
+    playerId: props.playerId,
+  })
+  const [data] = createResource(getOptions, PlayersApi.getPlayerMatches)
+  // FIXME
+  // const totalPages = () => Math.ceil((data()?.total ?? 1000) / count())
+  const totalPages = 1000
+
+  function updateHistory(options: ReturnType<typeof getOptions>, replace: boolean = false) {
+    const searchParams = new URLSearchParams(window?.location.search)
+    const { page } = options
+
+    if (page > 1) searchParams.set("page", page.toString())
+    else searchParams.delete("page")
+
+    const newLocation =
+      searchParams.toString().length > 0
+        ? `${window.location.pathname}?${searchParams.toString()}`
+        : window.location.pathname
+
+    if (searchParams.toString() == window.location.search || replace)
+      window.history.replaceState(options, "", newLocation)
+    else window.history.pushState(options, "", newLocation)
+  }
+
+  createEffect(() => {
+    if (!isBrowserNavigation()) updateHistory(getOptions())
+  })
+
+  onMount(() => {
+    window.addEventListener("popstate", (event) => {
+      try {
+        const state = event.state as Props
+        setIsBrowserNavigation(true)
+        setPage(state.page || 1)
+      } catch (e) {
+        console.error(e)
+      } finally {
+        window.setTimeout(() => setIsBrowserNavigation(false), 0)
+      }
+      return
+    })
+    updateHistory(getOptions(), true)
+    setIsBrowserNavigation(false)
+  })
+
+  return (
+    <div>
+      <Widget title="Match history" label="Closed Beta Ranked">
+        <Suspense fallback={<div>Loading...</div>}>
+          {/* FIXME */}
+          {false && <div class="my-6 text-center text-gray-400">No results found</div>}
+          <div class={isPending() ? "opacity-70" : ""}>
+            {data()?.matches.map((match) => (
+              <MatchPreview match={match} mainPlayerId={playerId} />
+            ))}
+          </div>
+
+          <div class="flex justify-center py-4">
+            {/* FIXME */}
+            {totalPages > 0 && (
```

Comment:
As discussed with @reneklacan, will do blind pagination

Reply:
Blind pagination fixed. 
@reneklacan we were wondering if being able to see your first matches might be a use case that user would want to have